crates/merod/src/cli/config.rs

```diff
@@ -44,53 +64,291 @@ impl FromStr for KeyValuePair {
 impl ConfigCommand {
     pub async fn run(self, root_args: &cli::RootArgs) -> EyreResult<()> {
         let path = root_args.home.join(&root_args.node_name);
-
         if !ConfigFile::exists(&path) {
             bail!("Node is not initialized in {:?}", path);
         }
 
         let path = path.join(CONFIG_FILE);
-
-        // Load the existing TOML file
         let toml_str = read_to_string(&path)
             .await
             .map_err(|_| eyre!("Node is not initialized in {:?}", path))?;
-
         let mut doc = toml_str.parse::<toml_edit::DocumentMut>()?;
 
-        // Update the TOML document
-        for kv in self.args.iter() {
-            let key_parts: Vec<&str> = kv.key.split('.').collect();
+        let (hints, edits): (Vec<_>, Vec<_>) = self
+            .args
+            .clone()
+            .into_iter()
+            .partition(|kv| kv.key.ends_with('?'));
+
+        if !hints.is_empty() {
+            return self.print_hints(&hints);
+        }
+
+        let mut changes_made = false;
+
+        if !edits.is_empty() {
+            for kv in &edits {
+                let key_parts: Vec<&str> = kv.key.split('.').collect();
+                let mut current = doc.as_item_mut();
+
+                for key in &key_parts[..key_parts.len() - 1] {
+                    current = &mut current[key];
+                }
+
+                let last_key = key_parts[key_parts.len() - 1];
+                let old_value = current[last_key].clone();
+                current[last_key] = Item::Value(kv.value.clone());
+
+                if old_value.to_string() != current[last_key].to_string() {
+                    changes_made = true;
+                }
```

Comment:
what happens if I specify any of these three:

1. `a.b.c=3` where `a.b` did not previously exist?

I suspect panic?

2.`a.b.c=3` where `a.b.c` was previously an object

it'll probably be overwritten, but then validated at a later step to be invalid, which is tolerable

3. `a.b.c=3 a.b.c.d=5`

`a.b.c` will probably discard `3` that was set, and make `a.b.c` an object with `.d` inside of it

Can you confirm these?